mesonbuild/backend/backends.py

```diff
@@ -737,6 +737,20 @@ def get_rpath_dirs_from_link_args(args: T.List[str]) -> T.Set[str]:
     def rpaths_for_non_system_absolute_shared_libraries(self, target: build.BuildTarget, exclude_system: bool = True) -> 'ImmutableListProtocol[str]':
         paths: OrderedSet[str] = OrderedSet()
         srcdir = self.environment.get_source_dir()
+
+        def add_path(libdir: str) -> None:
+            try:
+                commonpath = os.path.commonpath((libdir, srcdir))
+            except ValueError: # when paths are on different drives on Windows
+                commonpath = ''
+
+            if commonpath == srcdir:
+                rel_to_src = libdir[len(srcdir) + 1:]
+                assert not os.path.isabs(rel_to_src), f'rel_to_src: {rel_to_src} is absolute'
```

Comment:
What is the purpose of this assert? Is it expected this might happen?

Reply:
It was already there. I juste moved the code. I think I can remove it.

Reply:
Oh hmm, that's correct. We can leave it for now then.